File: 0007-整数反转/C++/reverseInt.cpp

// 07_整数反转_C++
//非好方法,小伙伴们有好的方法请补充,并在下方添加,谢谢!

class Solution {
public:
    int reverse(int inputInt) {
        int result = 0; 
        while (inputInt != 0) {
            int temp = inputInt % 10; //获得最后一位
            inputInt /= 10; //去除最后一位
            if (result > 214748364 || (result == 214748364 && temp > 7)) return 0;//正数溢出
            if (result < -214748364 || (result == -214748364 && temp < -8)) return 0;//负数溢出

            result = result * 10 + temp;//反转
        }
        return result;
    }
};

File: 0007-整数反转/Swift/reverseInt.swift
//
//  LJXReverseInt.swift
//  AuvGo
//
//  Created by 刘吉新 on 2019/5/24.
//  Copyright © 2019 com.auvgo.tmc. All rights reserved.
//

//写的比较low,有好方法的小伙伴欢迎指导,在该方法的下方,另写一个好的方法或指导意见,谢谢!

func reverse(_ x: Int) -> Int {
            var num = x
            var result = 0;
            while num != 0 {
                // 获得最后一个数
                let temp = num % 10
                // 去除最后一位  
                num /= 10  
                // 正数溢出检查
                if result > 214748364 || (result == 214748364 && temp > 7) {
                    return 0
                }
                // 负数溢出检查
                if result < -214748364 || (result == -214748364 && temp < -8) {
                    return 0
                }
                // 反转
                result = result * 10 + temp
            }
            return result
        }


